filter_plugins/oo_filters.py

```diff
@@ -128,14 +128,14 @@ def oo_select_keys(data, keys):
                 returns [1, 3]
         """
 
-        if not isinstance(data, dict):
-            raise errors.AnsibleFilterError("|failed expects to filter on a dict")
+        if not isinstance(data, dict) and not isinstance(data, object):
+            raise errors.AnsibleFilterError("|failed expects to filter on a dict or object")
```

Comment:
Ah, we're passing `hostvars` into `oo_select_keys` which is now an object (`collections.Mapping`). Since the `hostvars` object and dictionaries provide mapping interfaces, we can probably just check for `isinstance(data, collections.Mapping)` and that should work for python 2.6+.